soc/arm/nxp_lpc/lpc55xxx/soc.c

```diff
@@ -238,6 +238,17 @@ static int nxp_lpc55xxx_init(const struct device *arg)
 	/* restore interrupt state */
 	irq_unlock(oldLevel);
 
+
+	/* Set the Submodule Clocks for FlexPWM */
+#if defined(CONFIG_SOC_LPC55S36) && defined(CONFIG_PWM)
+	SYSCON->PWM0SUBCTL |=
+	  (SYSCON_PWM0SUBCTL_CLK0_EN_MASK | SYSCON_PWM0SUBCTL_CLK1_EN_MASK |
+	   SYSCON_PWM0SUBCTL_CLK2_EN_MASK);
+	SYSCON->PWM1SUBCTL |=
+	  (SYSCON_PWM1SUBCTL_CLK0_EN_MASK | SYSCON_PWM1SUBCTL_CLK1_EN_MASK |
+	   SYSCON_PWM1SUBCTL_CLK2_EN_MASK);
```

Comment:
Can this portion be added to clock_init function as it looks like clock setup for the PWM module.